framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
First, we should add a documentation string to the private parameters. Then we can addthem to the yaml dump and do what we want with them, including not show them in Peacock. However, if they are in the yaml dump than we can show them somewhere (i.e., in a developer information section of the website).